data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v2/debug/GetState.java

```diff
@@ -0,0 +1,123 @@
+import tech.pegasys.teku.api.schema.BeaconState;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.spec.SpecMilestone;
+
+public class GetState extends AbstractHandler implements Handler {
+  public static final String ROUTE = "/eth/v2/debug/beacon/states/:state_id";
+  private final ChainDataProvider chainDataProvider;
+
+  public GetState(final DataProvider dataProvider, final JsonProvider jsonProvider) {
+    this(dataProvider.getChainDataProvider(), jsonProvider);
+  }
+
+  public GetState(final ChainDataProvider chainDataProvider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.chainDataProvider = chainDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get state",
+      tags = {TAG_EXPERIMENTAL},
+      description =
+          "Returns full BeaconState object for given state_id.\n\n"
+              + "Use Accept header to select `application/octet-stream` if SSZ response type is required.",
+      pathParams = {@OpenApiParam(name = PARAM_STATE_ID, description = PARAM_STATE_ID_DESCRIPTION)},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = {
+              @OpenApiContent(type = HEADER_ACCEPT_JSON, from = GetStateResponseV2.class),
+              @OpenApiContent(type = HEADER_ACCEPT_OCTET)
+            }),
+        @OpenApiResponse(status = RES_BAD_REQUEST),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR),
+        @OpenApiResponse(status = RES_SERVICE_UNAVAILABLE, description = SERVICE_UNAVAILABLE)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    final Optional<String> maybeAcceptHeader = Optional.ofNullable(ctx.header(HEADER_ACCEPT));
+    final Map<String, String> pathParamMap = ctx.pathParamMap();
+    if (maybeAcceptHeader.orElse(HEADER_ACCEPT_JSON).equalsIgnoreCase(HEADER_ACCEPT_JSON)) {
+      final SafeFuture<Optional<BeaconState>> future =
+          chainDataProvider.getBeaconState(pathParamMap.get(PARAM_STATE_ID));
+      handleOptionalResult(ctx, future, this::handleJsonResult, SC_NOT_FOUND);
+    } else if (maybeAcceptHeader.orElse("").equalsIgnoreCase(HEADER_ACCEPT_OCTET)) {
+      final SafeFuture<Optional<StateSszResponse>> future =
+          chainDataProvider.getBeaconStateSsz(pathParamMap.get(PARAM_STATE_ID));
+      handleOptionalSszResult(
+          ctx, future, this::handleSszResult, this::resultFilename, SC_NOT_FOUND);
+    } else {
+      ctx.status(SC_BAD_REQUEST);
+      ctx.result(
+          BadRequest.badRequest(
+              jsonProvider, "Received an unsupported accept-header: " + maybeAcceptHeader.get()));
+    }
```

Comment:
Can we just default to JSON instead of responding with bad request (and update v1, it's been annoying me for a while).  `Accept` header is a polite request, not a requirement - if the requested type isn't available the server should just respond with it's default type which for us is JSON.